smqtk_indexing/utils/metrics.py

```diff
@@ -97,19 +97,28 @@ def cosine_similarity(i: np.ndarray, j: np.ndarray) -> float:
     :param i: Vector i
     :param j: Vector j
 
-    :return: Float similarity.
+    :return: Float or array of cosine similarity.
     """
-    # numpy.linalg.norm is Frobenius norm (vector magnitude)
-    # return numpy.dot(i, j) / (numpy.linalg.norm(i) * numpy.linalg.norm(j))
+    sum_axis = 1
+    if i.ndim == 1 and j.ndim == 1:
+        sum_axis = 0
 
-    # speed optimization, numpy.linalg.norm can be a bottleneck
-    # TODO(john.moeller): This can be made array-friendly
-    #   np.multiply(i, j).sum(axis) / (np.linalg.norm(i, i_axis) *
-    #       np.linalg.norm(j, j_axis))
-    return np.dot(i, j) / (np.sqrt(i.dot(i)) * np.sqrt(j.dot(j)))
```

Comment:
I couldn't get this suggestion to work with `linalg.norm`, so instead I calculated the magnitudes with `sqrt` and `sum`. I confirmed that this matches the result produced by scikit-learn's cosine similarity and scipy's cosine similarity functions. 

Alternatively we could just use the scikit-learn or scipy function.

Reply:
Sure, `scikit-learn` (and `scipy` by transitive dependency) are already required dep(s) so lets not reinvent the wheel if they already have a solution for this.